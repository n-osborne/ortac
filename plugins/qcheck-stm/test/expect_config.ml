open Ortac_qcheck_stm

let underline str =
  Fmt.pr "@[<v 0>%s@,%s@]@." str (String.map (Fun.const '=') str)

let pp = Reserr.pp (Fmt.pair Fmt.nop Config.dump) Fmt.stdout
let () = underline "test with correct cli arguments:"
let () = Config.init "lib.mli" "make 'a' 1" "(char, int) t" |> pp
let () = underline "test with non-existing sut type:"
let () = Config.init "lib.mli" "make 'a' 1" "absent" |> pp
let () = underline "test with ill-formed init_sut expression:"
let () = Config.init "lib.mli" "make (" "(char, int) t" |> pp
let () = underline "test with ill-formed type expression:"
let () = Config.init "lib.mli" "make () ()" "a-t" |> pp
let () = underline "test with non instantiated sut type parameter:"
let () = Config.init "lib.mli" "make () ()" "('a, int) t" |> pp
let () = underline "test with arrow type as sut:"
let () = Config.init "lib.mli" "make () ()" "int -> int" |> pp
