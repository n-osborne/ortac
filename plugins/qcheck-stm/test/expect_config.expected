test with correct cli arguments:
================================

sut_core_type: make 'a' 1; init_sut: (char, int) t

test with non-existing sut type:
================================

sut_core_type: make 'a' 1; init_sut: absent

test with ill-formed init_sut expression:
=========================================
Error: `make (' is not a well formed OCaml expression.

test with ill-formed type expression:
=====================================
Error: `a-t' is not a well formed type expression.

test with non instantiated sut type parameter:
==============================================
Error: Type parameter `'a' should be instantiated.

test with arrow type as sut:
============================
Error: The type `int -> int' given for the system under test is not supported.

